Core/Core/MobileCore.cs

```diff
@@ -48,35 +46,76 @@ public String ConfigFileName
 
         private static MobileCore instance;
 
+        private MobileCoreConfiguration mobileConfiguration;
+
+        private ServiceInstanceCache serviceInstanceCache = new ServiceInstanceCache();
+
+        /// <summary>
+        /// Gets the Mobile Core configuration.
+        /// </summary>
+        /// <value>The configuration.</value>
+        public MobileCoreConfiguration Configuration => this.mobileConfiguration;
+
+        /// <summary>
+        /// Gets HTTP service module of the MobileCore.
+        /// </summary>
+        public IHttpServiceModule HttpLayer { get; private set; }
+
         /// <summary>
         /// Holds MobileCore singleton instance. It's needed to initialize core before using this.
         /// </summary>
-        public static MobileCore Instance
-        {
-            get
-            {
-                if (instance == null)
-                {
-                    throw new InitializationException("Core is not initialized, don't forget to call MobileCore.init() before using this instance.");
-                }
+        public static MobileCore Instance => instance ?? throw new InitializationException("Core is not initialized, don't forget to call MobileCore.init() before using this instance.");
 
-                return instance;
-            }
-        }
+        /// <summary>
+        /// Returns the service of the given type, if present.
+        /// </summary>
+        /// <returns>The service.</returns>
+        /// <typeparam name="T">The type of the service to be returned.</typeparam>
+        public T GetService<T>() where T : IServiceModule => GetService<T>(typeof(T));
 
         /// <summary>
-        /// Gets HTTP service module of the MobileCore.
+        /// Returns the service identified by the given id if it exists.
         /// </summary>
-        public IHttpServiceModule HttpLayer
-        {
-            get; private set;
-        }
+        /// <returns>The service.</returns>
+        /// <param name="serviceId">Service identifier.</param>
+        /// <typeparam name="T">The type of the service to be returned.</typeparam>
+        public T GetService<T>(string serviceId) where T : IServiceModule => GetService<T>(typeof(T), GetServiceConfigurationById(serviceId));
 
-        private Dictionary<String, ServiceConfiguration> servicesConfig;
+        /// <summary>
+        /// Returns array of ServiceConfiguration, filtered by type
+        /// </summary>
+        /// <param name="type">type field of the configuration</param>
+        /// <returns>array of ServiceConfiguration</returns>
+        public ServiceConfiguration[] GetServiceConfigurationByType(String type) => this.mobileConfiguration.GetServiceConfigurationByType(type);
```

Comment:
Shouldn't it be named like `GetServiceConfigurationsByType` because it's possible to return more than one?